src/main/java/io/apicurio/datamodels/core/validation/rules/other/SecurityRequirementUniquenessValidationRule.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2019 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.datamodels.core.validation.rules.other;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import io.apicurio.datamodels.core.models.common.SecurityRequirement;
+import io.apicurio.datamodels.core.validation.ValidationRule;
+import io.apicurio.datamodels.core.validation.ValidationRuleMetaData;
+import io.apicurio.datamodels.openapi.v2.models.Oas20Operation;
+
+/**
+ * @author cfoskin@redhat.com
+ */
+public class SecurityRequirementUniquenessValidationRule extends ValidationRule {
+
+    /**
+     * Constructor.
+     * 
+     * @param ruleInfo
+     */
+    public SecurityRequirementUniquenessValidationRule(ValidationRuleMetaData ruleInfo) {
+        super(ruleInfo);
+    }
+
+    /**
+     * @param currentSecurityRequirement
+     * @param securityRequirementToCompare
+     * @return if duplicate security requirements are found
+     */
+    private boolean checkForDuplicateSecurityRequirement(SecurityRequirement currentSecurityRequirement,
+            SecurityRequirement securityRequirementToCompare) {
+        List<String> currentSecurityRequirementNames = currentSecurityRequirement
+                .getSecurityRequirementNames();
+        List<String> securityRequirementToCompareNames = securityRequirementToCompare
+                .getSecurityRequirementNames();
+        Collections.sort(currentSecurityRequirementNames);
+        Collections.sort(securityRequirementToCompareNames);// still to test if this transpiles
+        return currentSecurityRequirementNames.equals(securityRequirementToCompareNames);
+    }
+
+    /**
+     * @return if two sets of scopes match or not
+     */
+    private boolean checkIfScopesMatch() {
+        System.out.println("Checking if scopes match...");
+        Collections.sort(currentSecurityRequirementScopes);
+        Collections.sort(securityRequirementToCompareScopes);// need to still test if this transpiles
+        System.out.println(currentSecurityRequirementScopes);
+        System.out.println(securityRequirementToCompareScopes);
+        return currentSecurityRequirementScopes.equals(securityRequirementToCompareScopes);
+    }
+
+    private List<String> currentSecurityRequirementScopes = new ArrayList<String>();
+    private List<String> securityRequirementToCompareScopes = new ArrayList<String>();
```

Comment:
If you can make the visitor stateless that would be better.  I think the rules might be singletons (rather than created new each time validation is run) - so making them stateful could be a problem under load.  So I definitely recommend not using fields.  In this case it looks like you did this because you couldn't return two things from method `getScopes()`.  Of course, the fix in this case would have been to have `getScopes()` to return a `List` and then call it twice (once for each requirement).

Reply:
Yep sounds good. I done it for the reason you mentioned and will change it.